google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java

```diff
@@ -256,4 +259,48 @@ public SqmMultiTableInsertStrategy getFallbackSqmInsertStrategy(
       EntityMappingType entityDescriptor, RuntimeModelCreationContext runtimeModelCreationContext) {
     return NoOpSqmMultiTableInsertStrategy.INSTANCE;
   }
+
+  @Override
+  public String addSqlHintOrComment(
+      String sql,
+      QueryOptions queryOptions,
+      boolean commentsEnabled) {
+    if (hasCommentHint(queryOptions)) {
+      sql = applyHint(sql, queryOptions.getComment());
+    }
+    if (queryOptions.getDatabaseHints() != null && !queryOptions.getDatabaseHints().isEmpty()) {
+      sql = applyQueryHints(sql, queryOptions);
+    }
+    return super.addSqlHintOrComment(sql, queryOptions, commentsEnabled);
+  }
+
+  private String applyHint(String sql, String hint) {
```

Comment:
This and other below functions could be static

Reply:
Done